Makefile

```diff
@@ -13,14 +13,18 @@ include var.mk
 HETS_VERSION ?= $(shell printf `cat version_nr`)
 
 # We assume ghc 7+
-GHCVERSION := $(shell ghc --numeric-version)
+GHCVERSION := $(shell $(HC) --numeric-version)
+GHC_PARALLEL_FEATURE := $(shell expr $(shell $(HC) --numeric-version | cut -d '.' -f 1) \>= 8)
 NO_BIND_WARNING := -fno-warn-unused-do-bind
 HC_WARN := -Wall -fwarn-tabs \
   -fwarn-unrecognised-pragmas -fno-warn-orphans $(NO_BIND_WARNING)
 # uncomment HC_PROF for profiling (and comment out packages in var.mk)
 # call resulting binary with a final +RTS -p to get a file <binary>.prof
 #HC_PROF := -prof -auto-all -osuf p_o +RTS -K100m -RTS
 HC_OPTS += $(HC_WARN) $(HC_PROF)
+ifeq "$(GHC_PARALLEL_FEATURE)" "1"
+    HC_OPTS += -j
```

Comment:
Thanks for all the input.

The `-j` option *for GHC* is safe: GHC waits for all the dependent files to finish compiling before it starts compiling the target file. `-j` *for make* was not safe with the old Makefile (before you rewrote it) because it did not wait for the dependent targets. I don't know if this is fixed with your new Makefile.

Christian had no chance to include this flag for GHC, because he wasn't involved with Hets anymore when it was introduced with GHC 8.0.

When given only `-j` (instead of, for instance, `-j 2`), it uses all available CPU cores to compile Hets. Each core is assigned a different source file to compile.

Because there were (or still are?) problems with `-j` *for make*, I would rather not couple it with `-j` *for GHC`. Is there another option to let the user decide whether or not to use it? I would like to activate it by default because it significantly reduces compiling time (2 mins vs. 8 mins on my laptop).